lib/consolidate.js

```diff
@@ -91,13 +93,24 @@ function read(path, options, fn) {
   if (cached) return fn(null, str);
 
   // read
-  fs.readFile(path, 'utf8', function(err, str){
-    if (err) return fn(err);
-    // remove extraneous utf8 BOM marker
-    str = str.replace(/^\uFEFF/, '');
-    if (options.cache) readCache[path] = str;
-    fn(null, str);
-  });
+  if (nowReading[path]) {
+    if (!readListeners[path]) readListeners[path] = [fn];
+    else readListeners[path].push(fn);
+  }else{
+    nowReading[path] = true;
+    fs.readFile(path, 'utf8', function(err, str){
+      if (err) return fn(err);
```

Comment:
It doesn't seem obsolete to me unless I'm missing something.

On Wed, Jan 20, 2016 at 4:57 AM, Ruben Bridgewater <notifications@github.com
> wrote:

> In lib/consolidate.js
> <https://github.com/tj/consolidate.js/pull/171#discussion_r50250251>:
>
> > @@ -91,13 +93,24 @@ function read(path, options, fn) {
> >    if (cached) return fn(null, str);
> >
> >    // read
> > -  fs.readFile(path, 'utf8', function(err, str){
> > -    if (err) return fn(err);
> > -    // remove extraneous utf8 BOM marker
> > -    str = str.replace(/^\uFEFF/, '');
> > -    if (options.cache) readCache[path] = str;
> > -    fn(null, str);
> > -  });
> > +  if (nowReading[path]) {
> > +    if (!readListeners[path]) readListeners[path] = [fn];
> > +    else readListeners[path].push(fn);
> > +  }else{
> > +    nowReading[path] = true;
> > +    fs.readFile(path, 'utf8', function(err, str){
> > +      if (err) return fn(err);
>
> As far as I understand it this PR is obsolete, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tj/consolidate.js/pull/171/files#r50250251>.
>